packages/allure-js-commons/src/AllureCommandStep.ts

```diff
@@ -0,0 +1,267 @@
+import { AllureRuntime } from "./AllureRuntime";
+import {
+  Attachment,
+  AttachmentMetadata,
+  AttachmentOptions,
+  ContentType,
+  ExecutableItem,
+  LabelName,
+  LinkType,
+  ParameterOptions,
+  Stage,
+  Status,
+} from "./model";
+
+export type StepBodyFunction = (
+  this: AllureCommandStepExecutable,
+  step: AllureCommandStepExecutable,
+) => any | Promise<any>;
+
+export interface AllureCommandStep<T = AttachmentMetadata> {
+  name: string;
+
+  attachments: Attachment[];
+
+  metadata: T;
+
+  label(label: string, value: string): void | Promise<void>;
+
+  link(url: string, name?: string, type?: string): void | Promise<void>;
+
+  parameter(name: string, value: string, options?: ParameterOptions): void | Promise<void>;
+
+  epic(epic: string): void | Promise<void>;
+
+  feature(feature: string): void | Promise<void>;
+
+  story(story: string): void | Promise<void>;
+
+  suite(name: string): void | Promise<void>;
+
+  parentSuite(name: string): void | Promise<void>;
+
+  subSuite(name: string): void | Promise<void>;
+
+  owner(owner: string): void | Promise<void>;
+
+  severity(severity: string): void | Promise<void>;
+
+  tag(tag: string): void | Promise<void>;
+
+  issue(issue: string, url: string): void | Promise<void>;
+
+  tms(issue: string, url: string): void | Promise<void>;
+
+  attach(
+    name: string,
+    content: Buffer | string,
+    options: ContentType | string | AttachmentOptions,
+  ): void | Promise<void>;
+}
+
+// TODO: think about the class name
+export class AllureCommandStepExecutable implements AllureCommandStep {
+  name: string = "";
+
+  attachments: Attachment[] = [];
+
+  metadata: AttachmentMetadata = {};
+
+  constructor(name: string) {
+    this.name = name;
+  }
+
+  /**
+   * Recursively writes attachments from the given step and all it's children
+   * Mutates given step object!
+   */
+  static writeStepAttachments(runtime: AllureRuntime, step: ExecutableItem) {
+    if (step.attachments.length > 0) {
+      step.attachments.forEach((attachment) => {
+        const encoding = /(text|application)/.test(attachment.type) ? "utf8" : "base64";
+        const attachmentContent = Buffer.from(attachment.source, encoding);
```

Comment:
video to base64 sounds really exprensive and long 🤔 did we really need it?

Reply:
We can't send any binary data as buffer using ipc text messages. Could you propose something here?